doc/contribute/guidelines.rst

```diff
@@ -536,12 +536,28 @@ reference manuals, etc.
 Coding Style
 ============
 
+.. note::
+   Coding style is enforced on any new or modified code, but contributors are
+   not expected to correct the style on existing code that they are not
+   modifying.
+
+.. note::
+   For style aspects where the guidelines don't offer explicit guidance or
+   permit multiple valid ways to express something, new code must try to follow
```

Comment:
2 comments here:
- "new code" doesn't sound like a valid subject in this sentence. Rather than new code trying something, it's probably rather the PR author
- "must" (as in MUST) and "try" doesn't sound to me like they can be used in the same sentence -- this actually feels very much like a synonym for "SHOULD", for if one is expected to "try" to do something, we have to be quite flexible in terms of expectations. What I'm trying to say is that since there is no objective "right" way to implement this recommendation, I don't want it to be something that can justify a "you did it wrong" kind of feedback—especially when the contributor actually feels like they did *try*—let alone an actual nack.

Reply:
Agree with your objections, I think they are very valid. However, I do think the note serves a purpose, any suggestions as to how to rewrite it?

Reply:
@kartben has a fair point that the verb "try" suggests a human subject, and as such is not the best match for a subject like "the code". I think in this case we could just drop it and have either "must", "should", or "shall" there. 

> What I'm trying to say is that since there is no objective "right" way to implement this recommendation, I don't want it to be something that can justify a "you did it wrong" kind of feedback

I don't think the inability to specify something precisely is a valid reason to dismiss it here. I suppose we could extend this with a requirement for the maintainer/reviewer to be able to explain what the existing pattern/style in the existing code is when requesting such changes. Especially first time contributors to a piece of code are unlikely to see all such conventions/patterns/styles up front, and it's perfectly natural for there to be at least one feedback cycle in reviews to address them.

Reply:
@kartben how about the proposal from Discord:
>For style aspects where the guidelines don't offer explicit guidance or permit multiple valid ways to express something, contributors should follow the style of existing code in the tree, with higher importance given to "nearby" code (first look at the function, then the same file, then subsystem, etc).